onnxruntime/test/contrib_ops/attention_op_test_helper.cc

```diff
@@ -2908,6 +2908,188 @@ void GetCrossAttentionData_HeadSize16(AttentionTestData& data) {
   }
 }
 
+void GetCrossAttentionDataWithPast(AttentionTestData& data) {
+  data.hidden_size = 8;
+  data.v_hidden_size = 8;
+  data.num_heads = 2;
+  data.batch_size = 1;
+  data.sequence_length = 2;
+  data.kv_sequence_length = 3;
+  data.mask_type = AttentionMaskType::MASK_2D_KEY_PADDING;
+  data.key_padding_mask_data = {1, 1, 1};
+
+  data.skip_kernel_types = {
+      AttentionKernelType::AttentionKernel_TrtFlashAttention,
+      AttentionKernelType::AttentionKernel_Default,
+      AttentionKernelType::AttentionKernel_TrtFusedCrossAttention,
+      AttentionKernelType::AttentionKernel_TrtFusedAttention,
+      AttentionKernelType::AttentionKernel_CutlassMemoryEfficientAttention};
+
+  {
+    data.query_data = {
+      -0.10939738f, -0.11916742f, -0.23157823f, -0.12894472f,
+      -0.02661306f,  0.26251313f,  0.30725253f, -0.34759378f,
+      -0.11695808f, -0.13129434f, -0.17031054f, -0.14986445f,
+      -0.02826184f,  0.2797631f ,  0.27337456f, -0.44312602f
+    };
+  }
+  {
+    data.past_key_data = {
```

Comment:
I do not understand this case. When key is empty, and past_key has data, why not just use key to store past_key?